test/services/data_exports/create_service_test.rb

```diff
@@ -171,5 +178,79 @@ def setup
                           id: @workflow_execution1.id),
                    exception.result.message
     end
+
+    test 'create valid csv linelist data export and namespace_type group' do
+      sample32 = samples(:sample32)
+      sample33 = samples(:sample33)
+      group12 = groups(:group_twelve)
+
+      valid_params = {
+        'export_type' => 'linelist',
+        'export_parameters' => {
+          'ids' => [sample32.id, sample33.id],
+          'format' => 'csv',
+          'namespace_id' => group12.id,
+          'metadata_fields' => %w[metadatafield1 metadatafield2]
+        }
+      }
+
+      assert_difference -> { DataExport.count } => 1 do
+        DataExports::CreateService.new(@user, valid_params).execute
+      end
+    end
+
+    test 'create valid xlsx linelist data export and namespace_type project' do
+      valid_params = {
+        'export_type' => 'linelist',
+        'export_parameters' => {
+          'ids' => [@sample1.id, @sample2.id],
+          'format' => 'xlsx',
+          'namespace_id' => @project1.namespace.id,
+          'metadata_fields' => %w[metadatafield1 metadatafield2]
+        }
+      }
+
+      assert_difference -> { DataExport.count } => 1 do
+        DataExports::CreateService.new(@user, valid_params).execute
+      end
+    end
+
+    test 'cannot create sample export using samples the user is not authorized to export via group links' do
+      user = users(:david_doe)
+      group4 = groups(:david_doe_group_four)
+      invalid_params = {
+        'export_type' => 'sample',
+        'export_parameters' => {
+          'ids' => [@sample1.id, @sample2.id],
+          'namespace_id' => group4.id
+        }
+      }
+
+      assert_no_difference -> { DataExport.count } do
+        data_export = DataExports::CreateService.new(user, invalid_params).execute
+        assert_equal I18n.t('services.data_exports.create.invalid_export_samples'),
+                     data_export.errors.full_messages.first
+      end
+    end
+
+    test 'cannot create linelist export using samples user is not authorized to export via group links' do
+      user = users(:david_doe)
+      group4 = groups(:david_doe_group_four)
+      invalid_params = {
+        'export_type' => 'linelist',
+        'export_parameters' => {
+          'ids' => [@sample1.id, @sample2.id],
+          'namespace_id' => group4.id,
+          'format' => 'xlsx',
+          'metadata_fields' => %w[metadatafield1 metadatafield2]
+        }
+      }
+
+      assert_no_difference -> { DataExport.count } do
+        data_export = DataExports::CreateService.new(user, invalid_params).execute
+        assert_equal I18n.t('services.data_exports.create.invalid_export_samples'),
+                     data_export.errors.full_messages.first
+      end
+    end
   end
```

Comment:
With the changes above to set the data export to `error` status if a validation fails, can I get you to add some tests here to make sure that the correct status is set, and that a data export job is not created

Reply:
This is no longer necessary as per the refactor PR change comment by Eric previously (cee1dded6c2cefb862d31fa89ab59f93ba0f7bfe)